src/PostgREST/ApiRequest.hs

```diff
@@ -48,7 +49,7 @@ type RequestBody = BL.ByteString
 -- | Types of things a user wants to do to tables/views/procs
 data Action = ActionCreate | ActionRead
             | ActionUpdate | ActionDelete
-            | ActionInfo   | ActionInvoke
+            | ActionInfo   | ActionInvoke{isReadOnly :: Bool}
```

Comment:
An issue I see with the idea is that `iReadOnly` is only used for `ActionInvoke` and don't think it'll be good for separation of concerns to have that attribute in `ApiRequest`, all of his attributes seem more directly related to things that the user wants to get/set.

I've come up with a modification to the `relevantPayload` function to reduce the pattern matches, check my latest commit.